### PR TITLE
Replaced npm with yarn

### PR DESCRIPTION
**Pull request by clintonb**
Yarn is faster than npm and, through its use of lock files, ensures that everyone gets the same version of dependencies.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,14 +7,20 @@ ENV ENABLE_DJANGO_TOOLBAR 1
 
 WORKDIR /edx/app/credentials/credentials
 
+# Yarn key/repo
```
Comment (clintonb) [inline Dockerfile]: This will eventually be moved into the base images at https://github.com/edx/edx-docker-base.

**Comment by clintonb** [comment]
Thoughts on this change, @edx/learner, @edx/fedx-team, @gsong?

File: Dockerfile (hunk 1 above)
Comment (clintonb) [inline Dockerfile]: https://github.com/edx/edx-docker-base/pull/4

**Comment by arizzitano** [comment]
:+1: yay for yarn!

File: Dockerfile (hunk 1 above)
Comment (clintonb) [inline Dockerfile]: `yarn` is now in the base image.

**Pull request merged**